src/codegen/mod.rs

```diff
@@ -3560,6 +3598,49 @@ mod utils {
         result.extend(old_items.into_iter());
     }
 
+    pub fn prepend_block_header(
+        ctx: &BindgenContext,
+        result: &mut Vec<quote::Tokens>,
+    ) {
+        let use_block = if ctx.options().block_extern_crate {
+            quote! {
+                #![allow(improper_ctypes)]
+
+                extern crate block;
+            }
+        } else {
+            quote! {
+                #![allow(improper_ctypes)]
+
+                use block;
+            }
+        };
+
+        let void = helpers::ast_ty::raw_type(ctx, "c_void");
+        let def_block = quote! {
+            pub fn block<A, R, F>(closure: F) -> *const ::block::Block<A, R>
```

Comment:
I think this should not really belong in bindgen, fwiw. This ties bindgen to whatever `block`s API is. And if something in that crate changes, then bindgen needs a new release.

I think this should live in the `block` crate, if anything.

Reply:
agree, its not necessary, I will remove it